### PR TITLE
feat(api): Add statsPeriod and time range support to org events endpoint

### PR DESCRIPTION
**Pull request by macqueen**

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events.py [hunk 1]
```diff
@@ -62,6 +66,27 @@ def get(self, request, organization):
 
         now = timezone.now()
 
+        # default to last 90 days
+        end = now
+        start = now - timedelta(days=90)
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_events.py]: You could use `MAX_STATS_PERIOD` here. :bike:

File: src/sentry/api/endpoints/organization_events.py [hunk 2]
```diff
@@ -62,6 +66,27 @@ def get(self, request, organization):
...
+        elif request.GET.get('start') or request.GET.get('end'):
+            if not all([request.GET.get('start'), request.GET.get('end')]):
+                return Response({'detail': 'start and end are both required'}, status=400)
+            try:
+                start = parse_datetime_string(request.GET['start'])
+                end = parse_datetime_string(request.GET['end'])
+            except InvalidQuery as exc:
+                return Response({'detail': exc.message}, status=400)
+            if start > end:
+                return Response({'detail': 'start must be before end'}, status=400)
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_events.py]: This logic looks like the kind of thing that could live in a rest framework serializer if you wanted to trim down the endpoint logic.

File: tests/snuba/api/endpoints/test_organization_events.py [hunk 3]
```diff
@@ -123,3 +126,71 @@ def test_project_filtering(self):
...
+        url = '%s?statsPeriod=2h' % (url,)
+        response = self.client.get(url, format='json')
+
+        assert response.status_code == 200, response.content
+        assert len(response.data) == 1
+        assert sorted(map(lambda x: x['eventID'], response.data)) == sorted(
+            [
+                event_1.event_id,
+            ]
+        )
```
Comment (markstory) [inline tests/snuba/api/endpoints/test_organization_events.py]: You do this a couple times in the test, is it worth having an assertion method on the test class?

**Review by billyvg — COMMENTED** [review]
Should we add a test for when `statsPeriod` and `start`+`end` are defined?

File: src/sentry/api/endpoints/organization_events.py (hunk 2 above)
Comment (macqueen) [reply src/sentry/api/endpoints/organization_events.py]: we're probably going to replace this completely in the near future to support many more search terms, so i'd rather just leave how it is for now

**Review by macqueen — COMMENTED** [review]
@billyvg do you have an opinion on what we should do in that situation? error or just use statsPeriod? currently doing the latter, but can change

**Comment by macqueen** [comment]
@markstory just realized i'm going to need the same thing for another endpoint, so moved the logic into an external function at least, which seems somewhat cleaner at least

File: src/sentry/api/endpoints/organization_events.py [hunk 4]
```diff
@@ -13,10 +13,47 @@
...
+def get_date_range_from_params(params):
+    # Returns (start, end) or raises an `InvalidParams` exception
+    now = timezone.now()
+
+    end = now
+    start = now - MAX_STATS_PERIOD
+
+    stats_period = params.get('statsPeriod')
+    if stats_period is not None:
+        stats_period = parse_stats_period(params['statsPeriod'])
```
Comment (benvinegar) [inline src/sentry/api/endpoints/organization_events.py]: Nit: why not use the `stats_period` var here?

```suggestion
        stats_period = parse_stats_period(stats_period)
```

File: src/sentry/api/endpoints/organization_events.py [hunk 5]
```diff
@@ -13,10 +13,47 @@
...
+        if stats_period is None or stats_period < MIN_STATS_PERIOD or stats_period >= MAX_STATS_PERIOD:
+            raise InvalidParams('Invalid statsPeriod')
```
Comment (benvinegar) [inline src/sentry/api/endpoints/organization_events.py]: We don't already have an `InvalidParams` Exception class for other endpoints (or something similar)?

File: src/sentry/api/endpoints/organization_events.py [hunk 6]
```diff
@@ -61,7 +98,10 @@ def get(self, request, organization):
             conditions.append(
                 [['positionCaseInsensitive', ['message', "'%s'" % (query,)]], '!=', 0])
 
-        now = timezone.now()
+        try:
+            start, end = get_date_range_from_params(request.GET)
+        except InvalidParams as exc:
+            return Response({'detail': exc.message}, status=400)
```
Comment (benvinegar) [inline src/sentry/api/endpoints/organization_events.py]: oic

**Review by benvinegar — APPROVED** [review]
Given the scope of this is limited to org events, I think we can move forward and merge – @mattrobenolt can follow up if he has feedback.

**Comment by lynnagara** [comment]
Use parse_stats_period from https://github.com/getsentry/sentry/blob/master/src/sentry/utils/dates.py (we use that for discover)

**Comment by macqueen** [comment]
@lynnagara fixed in https://github.com/getsentry/sentry/pull/10140. i didn't want to wait for travis again since this was blocking that.

**Pull request merged**